packages/@react-stately/color/src/useHexColorFieldState.ts

```diff
@@ -0,0 +1,150 @@
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {Color} from './Color';
+import {ColorInput, HexColorFieldProps} from '@react-types/color';
+import {NumberFieldState} from '@react-stately/numberfield';
+import {useCallback, useState} from 'react';
+import {useColor} from './useColor';
+import {useControlledState} from '@react-stately/utils';
+
+export interface HexColorFieldState extends Omit<NumberFieldState, 'value' | 'setValue'> {
+  colorValue: Color,
+  setInputValue: (value: string) => void
+}
+
+export const defaultMinValue = new Color('#000000');
+export const defaultMaxValue = new Color('#FFFFFF');
+const minValueRange = [new Color('#000000'), new Color('#FFFFFE')];
+const maxValueRange = [new Color('#000001'), new Color('#FFFFFF')];
+
+export function useHexColorFieldState(
+  props: HexColorFieldProps
+): HexColorFieldState {
+  let {
+    minValue = defaultMinValue,
+    maxValue = defaultMaxValue,
+    step = 1,
+    value,
+    defaultValue,
+    onChange,
+    validationState
+  } = props;
+
+  let clampColor = (value: ColorInput, min: Color, max: Color) => {
+    try {
+      let color = typeof value === 'string' ? new Color(value) : value;
+      let colorInt = color.toHexInt();
+      if (colorInt < min.toHexInt()) { return min; }
+      if (colorInt > max.toHexInt()) { return max; }
+      return color;
+    } catch (err) {
+      return undefined;
+    }
+  };
+
+  let {color: minColor, colorInt: minColorInt} = useColor(clampColor(minValue, minValueRange[0], minValueRange[1]));
+  let {color: maxColor, colorInt: maxColorInt} = useColor(clampColor(maxValue, maxValueRange[0], maxValueRange[1]));
+
+  let initialValue = clampColor(value, minColor, maxColor);
+  let initialDefaultValue = clampColor(defaultValue, minColor, maxColor);
+  let [colorValue, setColorValue] = useControlledState<Color>(initialValue, initialDefaultValue, onChange);
+
+  let initialInputValue = (value || defaultValue) && colorValue ? colorValue.toString('hex') : '';
+  let [inputValue, setInputValue] = useState(initialInputValue);
+
+  let increment = () => {
+    setColorValue((prevColor: Color) => {
+      let prevColorInt = prevColor ? prevColor.toHexInt() : minColorInt;
+      let newColor = prevColor;
+      if (prevColorInt < maxColorInt) {
+        let newValue = `#${Math.min(prevColorInt + step, maxColorInt).toString(16).padStart(6, '0')}`;
+        newColor = new Color(newValue);
+      }
+      setInputValue(newColor.toString('hex'));
```

Comment:
Will change this! My reasoning here is that I wanted to rely on `Color` to decide how the color should be represented in string, instead of using a value calculated outside of it, like line 70.

Reply:
Resolved for increment and decrement